auction-server/src/auction/service/conclude_auction.rs

```diff
@@ -15,13 +15,55 @@ pub struct ConcludeAuctionInput<T: ChainTrait> {
     pub auction: entities::Auction<T>,
 }
 
+pub struct ConcludeAuctionWithStatusesInput<T: ChainTrait> {
+    pub auction:      entities::Auction<T>,
+    pub bid_statuses: Vec<(T::BidStatusType, entities::Bid<T>)>,
+}
+
 impl<T: ChainTrait> Service<T>
 where
     Service<T>: AuctionManager<T>,
 {
-    #[tracing::instrument(skip_all, fields(auction_id, tx_hash, bid_ids, bid_statuses))]
-    pub async fn conclude_auction(&self, input: ConcludeAuctionInput<T>) -> anyhow::Result<()> {
+    #[tracing::instrument(skip_all, fields(auction_id, bid_ids, bid_statuses))]
+    pub async fn conclude_auction_with_statuses(
+        &self,
+        input: ConcludeAuctionWithStatusesInput<T>,
+    ) -> anyhow::Result<()> {
         let mut auction = input.auction;
+        tracing::Span::current().record(
+            "bid_ids",
+            tracing::field::display(entities::BidContainerTracing(&auction.bids)),
+        );
+        tracing::Span::current().record("auction_id", auction.id.to_string());
+        tracing::Span::current().record("bid_statuses", format!("{:?}", input.bid_statuses));
+        join_all(input.bid_statuses.into_iter().map(|(status, bid)| {
+            self.update_bid_status(UpdateBidStatusInput {
+                bid:        bid.clone(),
+                new_status: status.clone(),
+            })
+        }))
+        .await;
+
+        if self
+            .repo
+            .get_in_memory_submitted_bids_for_auction(&auction)
+            .await
+            .is_empty()
+        {
+            self.repo
+                .conclude_auction(&mut auction)
+                .await
+                .map_err(|e| anyhow::anyhow!("Failed to conclude auction: {:?}", e))?;
+            self.repo.remove_in_memory_submitted_auction(auction).await;
+        }
+
+        Ok(())
+    }
+
+    /// This one concludes an auction by getting the auction transaction status from the chain.
+    #[tracing::instrument(skip_all, fields(auction_id, tx_hash, bid_ids))]
```

Comment:
I suggest removing this instrumentation.
tx_hash is a bit misleading now since auction can have multiple transactions.
The rest is traced inside the `conclude_auction_with_statuses` call.

Reply:
I removed the fields, but kept the instument itself.